org.testeditor.web.backend.persistence/src/test/java/org/testeditor/web/backend/persistence/DocumentResourceIntegrationTest.xtend

```diff
@@ -203,27 +205,36 @@ class DocumentResourceIntegrationTest extends AbstractPersistenceIntegrationTest
 		response.status.assertEquals(NOT_FOUND.statusCode)
 	}
 
+
+
 	private def Entity<String> stringEntity(CharSequence charSequence) {
 		return Entity.entity(charSequence.toString, MediaType.TEXT_PLAIN)
 	}
 
 	private def File write(String resourcePath, CharSequence charSequence) {
-		val file = getFile(resourcePath)
+		val file = getRemoteFile(resourcePath)
 		FileUtils.write(file, charSequence, StandardCharsets.UTF_8)
+		commitInRemoteRepository(resourcePath)
 		return file
 	}
 
 	private def String read(String resourcePath) {
-		val file = getFile(resourcePath)
+		val file = getRemoteFile(resourcePath)
 		file.exists.assertTrue('''File with path='«resourcePath»' does not exist.''')
 		return FileUtils.readFileToString(file, StandardCharsets.UTF_8)
 	}
 
-	private def File getFile(String resourcePath) {
+	private def File getLocalFile(String resourcePath) {
 		val userRoot = new File(workspaceRoot.root.path, userId)
 		return new File(userRoot, resourcePath)
 	}
 
+	private def File getRemoteFile(String resourcePath) {
+		val git = Git.open(remoteGitFolder.root)
+		git.reset.setMode(HARD).call // reset, so pushed changes are checked out into the working directory
```

Comment:
why do we need a reset here?

Reply:
Its somewhat easier to assert, e.g. file existance in the remote repo, on the working copy / file system, as opposed to checking the same by accessing the index. When changes are pushed to the remote, the changes are written into its index, but the remote's workspace is not sync'ed. The hard reset forces the working copy state to reflect the index state.